src/widgets/vitals/vitals-biometrics-form/vitals-biometrics-form.component.test.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { screen, render, act } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import VitalsAndBiometricForms from "./vitals-biometrics-form.component";
+import { savePatientVitals } from "../vitals-biometrics.resource";
+import { mockSessionDataResponse } from "../../../../__mocks__/session.mock";
+import { mockVitalsSignsConcept } from "../../../../__mocks__/vitals.mock";
+import { openmrsObservableFetch, openmrsFetch } from "@openmrs/esm-api";
+import { of } from "rxjs";
+
+const mockOpenmrsObservableFetch = openmrsObservableFetch as jest.Mock;
+const mockOpenmrsFetch = openmrsFetch as jest.Mock;
+const mockSavePatientVitals = savePatientVitals as jest.Mock;
+const mockCloseWorkspace = jest.fn();
+
+jest.mock("@openmrs/esm-api", () => ({
+  openmrsObservableFetch: jest.fn(),
+  openmrsFetch: jest.fn()
+}));
+
+jest.mock("../vitals-biometrics.resource", () => ({
+  savePatientVitals: jest.fn()
+}));
+
+window.getOpenmrsSpaBase = jest.fn();
+describe("<VitalsBiometricsForm/>", () => {
+  const mockVitalsBiometricsFormProps = { closeworkSpace: mockCloseWorkspace };
+  const promise = Promise.resolve(mockVitalsSignsConcept);
+  beforeEach(async () => {
+    mockOpenmrsFetch.mockImplementation(() => promise);
+    mockOpenmrsObservableFetch.mockImplementation(() =>
+      of(mockSessionDataResponse)
+    );
+    render(
+      <VitalsAndBiometricForms
+        closeWorkspace={mockVitalsBiometricsFormProps.closeworkSpace}
+      />
+    );
+    await act(() => promise);
+  });
+
+  afterEach(() => jest.restoreAllMocks());
+
+  it("should close form when cancel button is clicked", async () => {
+    const cancelButton = await screen.findByRole("button", { name: /Cancel/i });
+    userEvent.click(cancelButton);
+    expect(mockCloseWorkspace).toHaveBeenCalledTimes(1);
+  });
+
+  it("should save patient vitals and biometrics and close form", async () => {
+    const saveButton = screen.getByRole("button", { name: /Sign & Save/i });
+    userEvent.type(screen.getByTitle(/Respiration Rate/i), "70");
+    userEvent.type(screen.getByTitle(/Temp/i), "36.5");
+    userEvent.type(screen.getByTitle(/Oxygen Saturation/i), "93");
+    userEvent.type(screen.getByTitle(/pulse/i), "29");
+    userEvent.type(screen.getByTitle(/systolic/i), "120");
+    userEvent.type(screen.getByTitle(/diastolic/i), "80");
+    userEvent.type(screen.getByTitle(/Notes/i), "Patient vitals okay");
+    userEvent.type(screen.getByTitle(/Weight/i), "70");
+    userEvent.type(screen.getByTitle(/Height/i), "165");
+    userEvent.type(screen.getByTitle(/MUAC/i), "28");
+    mockSavePatientVitals.mockImplementation(() =>
+      Promise.resolve({ status: 201 })
+    );
+    userEvent.click(saveButton);
+    expect(mockSavePatientVitals).toHaveBeenCalled();
```

Comment:
I think asserting on the arguments with which this function was called, i.e.,
`expect(mockSavePatientVitals).toHaveBeenCalledWith()`, would make this a very valuable test. Right now it is not very valuable.

Note that you [don't even have to assert on all the parameters](https://codewithhugo.com/jest-specific-argument-parameter-assert/). The really important argument is the one that contains the data. Testing that will very concisely test a whole lot of logic distributed throughout the component.